File: java/08-04-2024/assignment4/src/loan/HomeLoan.java
package loan;

public class HomeLoan extends LoanClass {

    private double limit;

    public HomeLoan(double p, float pr) {
        super(p, pr);
        limit = 750000;
    }
    
    public float getRate() {
        double rate=0;
        if(principle > limit){
            rate = 0.01;
        }
        return (float) (principle < 20000000 ? 0.1+rate : 0.11+rate);
    }
}

File: dotnet/assignments/assignment3/Payroll/HomeLoan.cs
namespace Payroll
{
    class HomeLoan : Loan, Discountable
    {
      private double limit;
      public HomeLoan(double principle, double period) : base(principle, period) 
      {
        limit = 2500000;
      }
      public override double GetRate(){
          double rate = principle > 2000000 ? 0.11 : 0.1;
          if(principle > limit){
            rate = rate+0.01;
          }
          return rate;
      }
      
      public double getDiscount(){
        return GetEMI() * 0.10;
      }

    }

}